js/src/dapps/tokenreg/Inputs/Text/input-text.js

```diff
@@ -105,22 +105,22 @@ export default class InputText extends Component {
     const { validationType, contract } = this.props;
     const validation = validate(value, validationType, contract);
 
-    if (validation instanceof Promise) {
+    const loadingTimeout = setTimeout(() => {
       this.setState({ disabled: true, loading: true });
+    }, 50);
```

Comment:
Hmmm... It's more that it's not necessary to change the state twice if the request resolves in less than 50ms, no ?